src/infra/core/tests/tests/use_cases/test_create_dataset_from_snapshot_use_case.rs

```diff
@@ -87,51 +111,180 @@ async fn test_create_derived_dataset_fron_snapshot() {
         )
         .build();
 
-    harness.use_case.execute(snapshot_root).await.unwrap();
-    harness.use_case.execute(snapshot_derived).await.unwrap();
+    let options = Default::default();
+
+    let foo_create_res = harness
+        .use_case
+        .execute(snapshot_root, options)
+        .await
+        .unwrap();
+    let bar_create_res = harness
+        .use_case
+        .execute(snapshot_derived, options)
+        .await
+        .unwrap();
 
     assert_matches!(harness.check_dataset_exists(&alias_foo).await, Ok(_));
     assert_matches!(harness.check_dataset_exists(&alias_bar).await, Ok(_));
+
+    // Properties are set for multi-tenants only
+    assert_matches!(
+        harness
+            .rebac_service
+            .get_dataset_properties(&foo_create_res.dataset_handle.id)
+            .await,
+        Ok(props)
+            if props.is_empty()
+    );
+    assert_matches!(
+        harness
+            .rebac_service
+            .get_dataset_properties(&bar_create_res.dataset_handle.id)
+            .await,
+        Ok(props)
+            if props.is_empty()
+    );
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[tokio::test]
+async fn test_created_datasets_have_the_correct_visibility_attribute() {
+    let alias_private = DatasetAlias::new(None, DatasetName::new_unchecked("private"));
+    let alias_public = DatasetAlias::new(None, DatasetName::new_unchecked("public"));
+
+    let harness = CreateFromSnapshotUseCaseHarness::new(
+        Workspace::MultiTenant,
+        OutboxVariantVariant::OutboxImmediate,
+    );
+
+    {
+        let snapshot = MetadataFactory::dataset_snapshot()
+            .name(alias_private.clone())
+            .kind(DatasetKind::Root)
+            .push_event(MetadataFactory::set_polling_source().build())
+            .build();
+        let options = CreateDatasetUseCaseOptions {
+            dataset_visibility: DatasetVisibility::Private,
+        };
+
+        let private_create_res = harness.use_case.execute(snapshot, options).await.unwrap();
+
+        assert_matches!(
+            harness
+                .rebac_service
+                .get_dataset_properties(&private_create_res.dataset_handle.id)
+                .await,
+            Ok(props)
+                if props == [PropertyName::dataset_allows_public_read(false)]
+        );
```

Comment:
I don't think use case test is a right place to test rebac logic.
It would be fine to check that the visibility arguments sent to use case on the input are represented in the delivered outbox messages.

The use case, which we are testing here, does not care if Rebac Service exists!

Reply:
> I don't think use case test is a right place to test rebac logic.

Tests moved to `RebacServiceImpl` area -- Done